csharp/src/Ice/LocatorInfo.cs

```diff
@@ -229,6 +238,75 @@ internal void ClearCache(Reference reference)
             return (endpoints, cached);
         }
 
+        // Encoding represents the encoding used to send the getRegistry request.
+        internal async ValueTask<ILocatorRegistryPrx?> GetLocatorRegistryAsync(Encoding encoding)
+        {
+            if (_locatorRegistry == null)
+            {
+                ILocatorRegistryPrx? prx =
+                    await Locator.Clone(encoding: encoding).GetRegistryAsync().ConfigureAwait(false);
+
+                // The locator registry can't be located and we use ordered endpoint selection in case the locator
+                // returned a proxy with some endpoints which are preferred to be tried first.
+                _locatorRegistry = prx?.Clone(clearLocator: true, endpointSelection: EndpointSelectionType.Ordered);
+            }
+            return _locatorRegistry;
+        }
+
+        // Returns true if the time-to-live has been reached
```

Comment:
The comment looks wrong, `true` means that time elapsed since insertion is less or equal than ttl

Reply:
I removed the comment pending further refactoring.